sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -73,6 +73,10 @@ class JsonFunctionsSuite extends QueryTest with SharedSQLContext {
     checkAnswer(
       df.select($"key", functions.json_tuple($"jstring", "f1", "f2", "f3", "f4", "f5")),
       expected)
+
+    checkAnswer(
+      df.selectExpr("key", "json_tuple(jstring, 'f1', 'f2', 'f3', 'f4', 'f5')"),
```

Comment:
@yhuai Hello Yin, just debugged the code a little bit and trying hard to understand. In the json_tuple function in jsonExpression.scala, we compute the elementTypes as follows 

override def elementTypes: Seq[(DataType, Boolean, String)] = fieldExpressions.zipWithIndex.map {
    case (_, idx) => (StringType, true, s"c$idx")
  }

This name is then used while making the generator output in makeGeneratorOutput() in Analyzer. Do you think we should change this ?